packages/iTwinUI-react/src/core/Select/Select.tsx

```diff
@@ -255,25 +275,53 @@ export const Select = <T,>(props: SelectProps<T>): JSX.Element => {
           isSelected,
           onClick: () => {
             !option.disabled && onChange?.(option.value);
-            close();
+            !multi && close();
+          },
+          ref: (el: HTMLElement) => {
+            if (isSelected && !multi) {
+              el?.scrollIntoView();
+            }
           },
-          ref: (el: HTMLElement) => isSelected && el?.scrollIntoView(),
           role: 'option',
           ...option,
           ...menuItem.props,
         });
       });
     },
-    [itemRenderer, onChange, options, value],
+    [itemRenderer, multi, onChange, options, value],
   );
 
-  const selectedItem = React.useMemo(() => {
+  const selectedItems = React.useMemo(() => {
     if (value == null) {
       return undefined;
     }
-    return options.find((option) => option.value === value);
+    return Array.isArray(value)
```

Comment:
Maybe better check whether `multi` is true if for some reason someone decides to have single value that is an array.

Reply:
I have type guard, so user should not be able to pass array without `multiple`

Reply:
ammm... i dont know then :/ what you suggest

Reply:
Just check `multiple` 😀

Reply:
but then I need again some kind of a guard, cause types wont be recognized...

Reply:
Do you? Isn't `selectedItems` type `SelectOption[] | SelectOption`?

Reply:
and I see you already have type guard 
![image](https://user-images.githubusercontent.com/36186912/178496147-e9e31b56-a333-4cda-9ee7-0d926c80a497.png)

Reply:
But here we are talking about value :/

Reply:
Oh, now I understand.

Reply:
Or can you have a typeguard with generics where you pass `multiple` and other field that you care about.

Reply:
added